client/payment-methods/generated-map.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal dependencies
+ */
+import type { PaymentMethodMapEntry } from '../payment-methods-map';
```

Comment:
I'm not exactly sure how this is compiling (maybe because it's just importing the TS type?), but there seem to be a circular dependency, here. `client/payment-methods/generated-map.ts` is importing `client/payment-methods-map.tsx`. But `client/payment-methods-map.tsx` is also importing `client/payment-methods/generated-map.ts`.

Maybe `PaymentMethodMapEntry` should be separated in a different file.

Reply:
Thanks, good catch! Resolved this in a16c8f3d8